aten/src/ATen/native/cuda/Unique.cu

```diff
@@ -53,15 +54,23 @@ template <typename scalar_t>
     int64_t num_out = thrust::unique(policy, output_data, output_data + num_inp) - output_data;
     output.resize_(num_out);
 
+    Tensor counts = at::empty({0}, self.options().dtype(kLong));
+    if (return_counts) {
```

Comment:
You'd need to condition computation of inverse_indices on return_counts too, otherwise if return_counts is true and inverse_indices is false, this won't work

Reply:
Alternatively, you can do thrust::unique_by_key with counting iterator as a key, and thrust::adjacent_differences to get counts, that will probably have better worst case performance for cases where you have a lot of equal elements and there would be a lot of atomic collisions in scatter_add_, and you also won't compute inverse_indices that you might not need.